test/object-store/migrations.cpp

```diff
@@ -1050,6 +1050,368 @@ TEST_CASE("migration: Automatic") {
+            REQUIRE(child_table->size() == 1);
+            REQUIRE(parent_table->size() == 0);
+
+            REQUIRE_NOTHROW(realm->update_schema(set_table_type(schema, "child_table", ObjectType::Embedded), 2,
+                                                 [](auto, auto, auto&) {}));
+
+            REQUIRE(realm->schema_version() == 2);
+            REQUIRE(child_table->is_embedded());
+            REQUIRE(child_table->size() == 0);
+            REQUIRE(parent_table->size() == 0);
+        }
+        SECTION("change table from top-level to embedded, migration allowed, embedded object with 1 incoming link. "
+                "Resolve automatic "
+                "should not be triggered") {
+            InMemoryTestFile config;
+            config.automatic_handle_backlicks_in_migrations = true;
+            Schema schema = {
+                {"child_table",
+                 {
+                     {"value", PropertyType::Int},
+                 }},
+                {"parent_table",
+                 {
+                     {"child_property", PropertyType::Object | PropertyType::Nullable, "child_table"},
+                 }},
+            };
+            auto realm = Realm::get_shared_realm(config);
+            realm->update_schema(schema, 1);
+            auto child_table = ObjectStore::table_for_object_type(realm->read_group(), "child_table");
+            auto parent_table = ObjectStore::table_for_object_type(realm->read_group(), "parent_table");
+            realm->begin_transaction();
+            Obj child_object = child_table->create_object();
+            child_object.set("value", 42);
+            auto child_object_key = child_object.get_key();
+            parent_table->create_object().set_all(child_object_key);
+            realm->commit_transaction();
+            REQUIRE_FALSE(child_table->is_embedded());
+            REQUIRE(child_table->size() == 1);
+            REQUIRE(parent_table->size() == 1);
+
+            REQUIRE_NOTHROW(realm->update_schema(set_table_type(schema, "child_table", ObjectType::Embedded), 2,
+                                                 [](auto, auto, auto&) {}));
+
+            REQUIRE(realm->schema_version() == 2);
+            REQUIRE(child_table->is_embedded());
+            REQUIRE(child_table->size() == 1);
+            REQUIRE(parent_table->size() == 1);
+        }
+        SECTION("change table to embedded - multiple incoming links - resolved automatically + copy set, dictionary "
+                "verification") {
+            InMemoryTestFile config;
+            config.automatic_handle_backlicks_in_migrations = true;
+            Schema schema = {
+                {
+                    "child_table",
+                    {{"value", PropertyType::Int},
+                     {"value_dict", PropertyType::Dictionary | PropertyType::Int},
+                     {"links_dict", PropertyType::Dictionary | PropertyType::Object | PropertyType::Nullable,
+                      "target"},
```

Comment:
If you also add a Dictionary of type Mixed and add an ObjLink value, you will find the problem I mentioned about ObjLinks above.
`{"mixed_dict", PropertyType::Dictionary | PropertyType::Mixed | PropertyType::Nullable},`

Reply:
test added.